app/src/ui/diff/index.tsx

```diff
@@ -952,6 +974,65 @@ export class Diff extends React.Component<IDiffProps, {}> {
     return null
   }
 
+  private renderLargeTextDiff() {
+    return (
+      <div className="panel empty large-diff">
+        <img src={NoDiffImage} />
+        <p>
+          The diff is too large to be displayed by default.
+          <br />
+          You can try to show it anyways, but performance may be negatively
+          impacted.
+        </p>
+        <Button onClick={this.onShowLargeDiff}>Show diff</Button>
+      </div>
+    )
+  }
+
+  private renderDiffTooLarge() {
+    return (
+      <div className="panel empty large-diff">
+        <img src={NoDiffImage} />
+        <p>The diff is too large to be displayed.</p>
+      </div>
+    )
+  }
+
+  private renderLargeText(diff: ILargeTextDiff) {
+    if (diff.hunks == null || diff.text == null) {
```

Comment:
we do this check in the switch statement in `renderDiff` right before we call this method. if there's any concern then we can just have all `DiffType.LargeText` call the `this.renderLargeText` method (or something similarly titled) and branch inside.

Reply:
There is no concern, just a design decision. 

Since `hunks` and `text` are optional properties on `ILargeTextDiff` and `ITextDiff` does not allow for optional values, I have to prove to the TSC that those properties are not `null`. The two options I had was to use a conditional statement or to use the `!` operator and a comment e.g. `diff.hunks!  \\guaranteed to be set at this point`. I chose the former because you'd have to look for that condition in another method and I felt it was better off inside the method.